feat: Ability to (not) show vcs ignored files

Guergeiro opened the pull request:
Closes #4.

By default, the `:Select projectfile` command will not show the ignored
files from your version control system, unless you explicitly change
this behaviour with `let g:select_no_ignore_vcs = 1`.

Guergeiro committed:
feat: Ability to (not) show vcs ignored files

By default, the `:Select projectfile` command will not show the ignored
files from your version control system, unless you explicitly change
this behaviour with `let g:select_no_ignore_vcs = 1`.

habamax commented:
Thx for PR!

Could you also:
1. add help section/topic/smth for that setting
2. I think by default it should preserve previous behavior -- no_ignore_vcs = 1

habamax commented:
What about:

```
"""
""" Select projectfile
"""
let s:select.projectfile = {}
if executable('fd')
    let s:select.projectfile.data = {"job": "fd --path-separator / --type f --hidden --follow --exclude .git " .. get(g:, "select_no_ignore_vcs", "--no-ignore-vcs")}
elseif executable('rg')
    let s:select.projectfile.data = {"job": "rg --path-separator / --files --hidden --glob !.git " .. get(g:, "select_no_ignore_vcs", "--no-ignore-vcs")}
elseif !has("win32")
    let s:select.projectfile.data = {"job": "find -type f -not -path \"*/.git/*\""}
else
    let s:select.projectfile.data = ""
endif
```

Then one would have to make `let g:select_no_ignore_vcs = ""` to switch it off.


or 

```
"""
""" Select projectfile
"""
let s:select.projectfile = {}
let s:no_ignore_vcs = get(g:, "select_no_ignore_vcs", v:true) ? " --no-ignore-vcs" : ""
if executable('fd')
    let s:select.projectfile.data = {"job": "fd --path-separator / --type f --hidden --follow --exclude .git" .. s:no_ignore_vcs}
elseif executable('rg')
    let s:select.projectfile.data = {"job": "rg --path-separator / --files --hidden --glob !.git" .. s:no_ignore_vcs}
elseif !has("win32")
    let s:select.projectfile.data = {"job": "find -type f -not -path \"*/.git/*\""}
else
    let s:select.projectfile.data = ""
endif
```

Guergeiro commented:
I feel like a `true` or `false` flag would be better honestly. But if you prefer using a `string` variable, hey it's your project 😄
What do you think about the opposite then? The user would have to do `let g:select_no_ignore_vcs = 0`, that is, by default we check with `--no-ignore-vcs`, but if the user ***opts out*** we don't append `--no-ignore-vcs`.

habamax commented:
> I feel like a `true` or `false` flag would be better honestly. But if you prefer using a `string` variable, hey it's your project 😄

Boolean would be better indeed.

> What do you think about the opposite then? The user would have to do `let g:select_no_ignore_vcs = 0`, that is, by default we check with `--no-ignore-vcs`, but if the user _**opts out**_ we don't append `--no-ignore-vcs`.

I am fine with it --that is actually the second variant in my comment.

Could you try it and if it works for you, update the PR (also adding a bit of text to `doc/select.txt`)?

Guergeiro committed:
refactor: Changed logic to the negative

In the previous commit the user would use `let g:select_no_ignore_vcs =
1` to OPT IN and allow ignored files to be selected. With this commit,
the logic is changed so that ignored files are shown by default and the
user OPTS OUT of this behaviour with `let g:select_no_ignore_vcs = 0`.

The code around `autoload/select/def.vim` that concerns this feature was
also improved.

Updated `README.adoc` to account for the changes.

Updated `doc/select.txt` to account for the changes.

Guergeiro commented:
@habamax done. If there's some quick thing you want to change (specially wording around documentation), you can edit my PR 😃

habamax committed:
Update select.txt

habamax committed:
Update README.adoc

habamax commented:
Thx!
